File: Frontend/src/DTOs/multipleChoiceOption.ts
import type { BaseDTO } from './base';

export interface MultipleChoiceOptionDTO {
  text: string;
  isCorrect: boolean;
  order: number;
}

export interface MultipleChoiceOptionResponseDTO extends BaseDTO {
  text: string;
  isCorrect: boolean;
  order: number;
  assignmentId: number;
}

File: Backend/Models/MultipleChoiceOption.cs
using System.ComponentModel.DataAnnotations;

namespace TechnicalSkill.Models;

public class MultipleChoiceOption
{
    [Key]
    public int Id { get; set; }

    [Required]
    [StringLength(200)]
    public string Text { get; set; } = string.Empty;

    public bool IsCorrect { get; set; }

    [Required]
    public int AssignmentId { get; set; }
    public Assignment? Assignment { get; set; }

    public int Order { get; set; } // To maintain the order of options
}
